src/main/java/frc/robot/subsystems/conveyor/Conveyor.java

```diff
@@ -0,0 +1,236 @@
+            return DriverStation.Alliance.Red;
+        } else if (resultColor == Constants.Conveyor.BLUE) {
+            return DriverStation.Alliance.Blue;
+        } else {
+            return DriverStation.Alliance.Invalid;
+        }
+    }
+
+    /**
+     * @return the amount of balls inside the cargo.
+     */
+    public int getCargoCount() {
+        return position.size();
+    }
+
+    /**
+     * Set motor power.
+     *
+     * @param power the power to the motor [-1, 1].
+     */
+    public void setPower(double power) {
+        motor.set(power);
+    }
+
+    /**
+     * open the "flap" - solenoid
+     */
+    public void openFlap() {
+//        flap.set(false);
+    }
+
+    /**
+     * closes the "flap" - solenoid
+     */
+    public void closeFlap() {
+//        flap.set(true);
+    }
+
+    /**
+     * toggles the "flap" - solenoid
+     */
+    public void toggle() {
+//        flap.toggle();
+    }
+
+    private void updateActualBallPositions() throws InterruptedException {
+        var colorIntake = getColor();
+        boolean currentPostFlapBeamInput = postFlapBeam.get();
+        SmartDashboard.putString("alliance", colorIntake.name());
+        double power = 1;
+//        double power = motor.getMotorOutputPercent();
+        if (currentPostFlapBeamInput && !lastPostFlapBeamInput && power > 0) {
+            if(!position.toArray()[0].equals(DriverStation.Alliance.Invalid.name())) {
+                position.poll();
+                var temp = position.take();
+                position.add(DriverStation.Alliance.Invalid.name());
+                position.add(temp);
+            } else {
+                position.poll();
+                position.poll();
+                position.add(DriverStation.Alliance.Invalid.name());
+                position.add(DriverStation.Alliance.Invalid.name());
+            }
+        }
+        if (colorIntake != lastSeenColor && !colorIntake.equals(DriverStation.Alliance.Invalid)) {
+            if (power < 0) {
+                if (position.remainingCapacity() == 0) {
+                    position.poll();
+                }
+                position.offer(colorIntake.name());
+            } else if (power > 0) {
+                position.remove();
+                position.offer(colorIntake.name());
+            }
+        }
+        lastPostFlapBeamInput = currentPostFlapBeamInput;
+        lastSeenColor = colorIntake;
+        SmartDashboard.putString("lastSeen", lastSeenColor.name());
+    }
+
+    private Queue getQueue() {
+        /*
+        Case #1  queue = [None, None]
+        Case #2  queue = [None, Alliance]
+        Case #3  queue = [Alliance, None]
+        Case #4  queue = [Alliance, Alliance]
+        Case #5  queue = [Alliance, Opponent]
+        Case #6  queue = [None, Opponent]
+        Case #7  queue = [Opponent, None]
+        Case #8  queue = [Opponent, Opponent]
+        Case #9  queue = [Opponent, Alliance]

```

Comment:
Document the order of elements in the queue – is the first Cargo in the queue near the intake or the shooter?